addons/im_livechat/models/ir_binary.py

```diff
@@ -0,0 +1,23 @@
+from odoo import models
+
+
+class IrBinary(models.AbstractModel):
+    _inherit = 'ir.binary'
+
+    def _find_record_check_access(self, record, access_token, field):
+        """ Custom access check allowing to retrieve an operator's avatar.
+
+        Here, we assume that if you are a member of at least one
+        im_livechat.channel, then it's ok to make your avatar publicly
+        available.
+
+        We also make the chatbot operator avatars publicly available. """
+        if record._name == "res.partner" and field == "avatar_128":
+            is_livechat_member = bool(self.env['im_livechat.channel'].sudo().search_count([('user_ids', 'in', record.sudo().user_ids.ids)]))
+            if is_livechat_member:
+                return record.sudo()
+            # we don't put chatbot operators as livechat members (because we don't have a user_id for them)
```

Comment:
This comment is irrelevant/obsolete. In particular, we also don't put them as livechat member because we definitely don't want them to be automatically assigned to visitors as if they were an actual operator.